src/test/test_psbt.py

```diff
@@ -1,185 +1,291 @@
+        self.assertEqual(self.to_base64(psbt), 'cHNidP8BAgR7AAAAAQQBAAEFAQABBgEDAfsEAgAAAAA=')
+
+        # Create an input
+        tx_input = pointer(wally_tx_input())
+
+        txhash, txhash_len = make_cbuffer('e7f25add4560021c77c4944f92739025fddbf99816d79c06d219268ca9f4b7e7')
+        ret = wally_tx_input_init_alloc(txhash, txhash_len, 5, 6, b'\x59', 1, None, tx_input)
+        self.assertEqual(WALLY_OK, ret)
+        ret = wally_psbt_add_tx_input_at(psbt, 0, 0, tx_input)
+        self.assertEqual(WALLY_OK, ret)
+        ret, base64 = wally_psbt_to_base64(psbt, 0)
+        self.assertEqual(WALLY_OK, ret)
+        self.assertEqual('cHNidP8BAgR7AAAAAQQBAQEFAQABBgEDAfsEAgAAAAABDiDn8lrdRWACHHfElE+Sc5Al/dv5mBbXnAbSGSaMqfS35wEPBAUAAAABEAQGAAAAAA==', base64)
+
+        ret = wally_psbt_input_set_required_lockheight(psbt.contents.inputs[0], 499999999)
+        self.assertEqual(WALLY_OK, ret)
+        ret, base64 = wally_psbt_to_base64(psbt, 0)
+        self.assertEqual(WALLY_OK, ret)
+        self.assertEqual('cHNidP8BAgR7AAAAAQQBAQEFAQABBgEDAfsEAgAAAAABDiDn8lrdRWACHHfElE+Sc5Al/dv5mBbXnAbSGSaMqfS35wEPBAUAAAABEAQGAAAAARIE/2TNHQA=', base64)
+
+        tx_output = pointer(wally_tx_output())
+
+        wally_tx_output_init_alloc(1234, b'\x59\x59', 2, tx_output)
+        self.assertEqual(WALLY_OK, ret)
+        ret = wally_psbt_add_tx_output_at(psbt, 0, 0, tx_output)
+        self.assertEqual(WALLY_OK, ret)
+
+        ret, base64 = wally_psbt_to_base64(psbt, 0)
+        self.assertEqual(WALLY_OK, ret)
+        self.assertEqual('cHNidP8BAgR7AAAAAQQBAQEFAQEBBgEDAfsEAgAAAAABDiDn8lrdRWACHHfElE+Sc5Al/dv5mBbXnAbSGSaMqfS35wEPBAUAAAABEAQGAAAAARIE/2TNHQABAwjSBAAAAAAAAAEEAllZAA==', base64)
+
+    def test_invalid_args(self):
+        """Test invalid arguments to various PSBT functions"""
+        psbt = pointer(wally_psbt())
+
+        # init
+        cases = [
+            (1, 0, 0, 0, 0, psbt), # Invalid version
+            (0, 0, 0, 0, 0xff, psbt), # Invalid flags
+            (2, 0, 0, 0, 0xff, psbt), # Invalid flags (v2)
+            (0, 0, 0, 0, INIT_PSET, psbt), # v0 PSET
+            (0, 0, 0, 0, 0, None), # NULL dest
+        ]
+        for args in cases:
+            self.assertEqual(WALLY_EINVAL, wally_psbt_init_alloc(*args))
+
+        # psbt_from_base64
+        src_base64 = JSON['valid'][0]['psbt']
+        for args in [(None,       psbt),  # NULL base64
+                     ('',         psbt),  # Invalid flags
+                     (src_base64, None)]: # NULL dest
+            self.assertEqual(WALLY_EINVAL, wally_psbt_from_base64(*args))
+
+        self.assertEqual(WALLY_OK, wally_psbt_from_base64(JSON['valid'][0]['psbt'], psbt))
+
+        # psbt_clone_alloc
+        clone = pointer(wally_psbt())
+        for args in [(None, 0x0, clone), # NULL src
+                     (psbt, 0x1, clone), # Invalid flags
+                     (psbt, 0x0, None)]: # NULL dest
+            self.assertEqual(WALLY_EINVAL, wally_psbt_clone_alloc(*args))
```

Comment:
Please add at least a basic test for `wally_psbt_blind`

Reply:
There are no test vectors for this call yet, and the test setup is non-trivial, so a meaningful value test is not possible in a reasonable time frame IMO. If I am granted further time to work on this I will add role tests reconciled with Elements. In the meantime the external users like https://github.com/jgriffiths/wally_swap_test will have to suffice.